src/index.ts

```diff
@@ -167,6 +167,10 @@ export function presetTheme<T extends Record<string, any>>(options: PresetThemeO
               line = line.replace(/\..*?\s(.*\{)/, '$1')
             }
             return line
+          }).sort((a, b) => {
+            if (a.match(/^:root|^@media|^\}/)?.length)
```

Comment:
you need to get the light's selector and use it to sort the theme.

Reply:
Yes, that's a problem I didn't catch, I'll fix it.